manifests/vrrp/instance.pp

```diff
@@ -195,17 +201,72 @@
   $multicast_source_ip                                              = undef,
   $unicast_source_ip                                                = undef,
   $unicast_peers                                                    = undef,
+  $collect_unicast_peers                                            = false,
   $dont_track_primary                                               = false,
   $use_vmac                                                         = false,
   $vmac_xmit_base                                                   = true,
   Boolean $native_ipv6                                              = false,
-
 ) {
   $_name = regsubst($name, '[:\/\n]', '')
 
+  if (!is_integer($priority) or ($priority + 0) < 1 or ($priority + 0) > 254) {
+    fail('priority must be an integer 1 >= and <= 254')
+  }
+
+  if (!is_integer($virtual_router_id) or ($virtual_router_id + 0) < 1 or ($virtual_router_id + 0) > 255) {
+    fail('virtual_router_id must be an integer >= 1 and <= 255')
+  }
+
+  if $unicast_source_ip != undef and ! $collect_unicast_peers {
+    validate_ip_address($unicast_source_ip)
+  }
+
+  if $unicast_peers != undef {
```

Comment:
here we could use `Optional[Variant[String,Array[]]]`

If I understood the code correctly, unicast_peers is a single ip address or an array of ip addresses? In that case we could even do

`Variant[Array[Stdlib::IP::Address], Stdlib::IP::Address] $unicast_peers = []`